File: reflekt-java/src/main/java/se/jensim/reflekt/ReflektConf.java
package se.jensim.reflekt;

public class ReflektConf {

    private final String packageFilter;
    private final boolean includeNestedJars;

    private ReflektConf(Builder builder) {
        packageFilter = builder.packageFilter;
        includeNestedJars = builder.includeNestedJars;
    }

    public static Builder builder() {
        return new Builder();
    }

    /**
     * Filter the class files found according to this filter, it is used as a starts-with expression
     */
    public String getPackageFilter() {
        return packageFilter;
    }

    /**
     * Do deep scan into nested jars
     */
    public boolean isIncludeNestedJars() {
        return includeNestedJars;
    }

    public static class Builder {

        private String packageFilter = "";
        private boolean includeNestedJars = false;

        private Builder() {
        }

        /**
         * Filter the class files found according to this filter, it is used as a starts-with expression
         */
        public Builder setPackageFilter(String packageFilter) {
            this.packageFilter = packageFilter;
            return this;
        }

        /**
         * Do deep scan into nested jars
         */
        public Builder setIncludeNestedJars(boolean includeNestedJars) {
            this.includeNestedJars = includeNestedJars;
            return this;
        }

        public ReflektConf build() {
            return new ReflektConf(this);
        }
    }
}

File: reflekt-kotlin/src/main/kotlin/se/jensim/reflekt/RefleKtConf.kt
package se.jensim.reflekt

class RefleKtConf {

    /**
     * Filter the classes available for reflection, used as a class-canonicalName prefix.
     * A class named com.example.Myclass can then be matched with "com.exa", but not with "om.example".
     */
    var packageFilter: String = ""

    /**
     * Configuration container for the class file locators
     * @see ClassFileLocator
     */
    var classFileLocatorConf: ClassFileLocatorConf = ClassFileLocatorConf()

    /**
     * Configuration container for the class file locators
     * @see ClassFileLocator
     */
    fun classFileLocatorConf(conf: ClassFileLocatorConf.() -> Unit) {
        classFileLocatorConf.also(conf)
    }

    /**
     * Configuration container for the class file locators
     * @see ClassFileLocator
     */
    class ClassFileLocatorConf {

        /**
         * By default RefleKt will look recursively in classpathDirectories and inside the running jar, if applicable.
         * These are added to the extraClassFileLocator list
         * @see se.jensim.reflekt.internal.JarFileClassLocator
         * @see se.jensim.reflekt.internal.ClassPathClassFileLocator
         */
        var disableAllDefaultClassFileLocators = false

        /**
         * Instruction to the default classFileLocators weather or not to deep scan jars inside the running jar or jars on the classpath recursively.
         */
        var includeNestedJars = false

        /**
         * Made your own ClassFileLocator? Append it here.
         */
        val extraClassFileLocator = mutableListOf<ClassFileLocator>()
    }
}
